desc/plotting.py

```diff
@@ -3843,3 +3845,144 @@ def plot_logo(save_path=None, **kwargs):
         fig.savefig(save_path, facecolor=fig.get_facecolor(), edgecolor="none")
 
     return fig, ax
+
+
+def plot_adiabatic_invariant(
+    eq, rhos=None, alphas=None, num_pitch=None, pitch_idx=-1, mode="single-surface"
+):
+    """Plot the second adiabatic invariant J_|| (parallel action).
+
+    This function visualizes the parallel action invariant J_|| in either of two modes:
```

Comment:
I could see possibly splitting this into 2 different functions for the cases below, similar to how we have our existing `plot_section` vs `plot_1d/2d/3d` etc.